source/developer/oauth2-applications.rst

```diff
@@ -0,0 +1,80 @@
+
+3 - Set `Enable OAuth 2.0 Service Provider <https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider>`_ to **True**.
+
+4 - (Optional) If you would like to allow external applications to post with customizable usernames and profile pictures, then set `Enable integrations to override usernames <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames>`_ and `Enable integrations to override profile picture icons <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-profile-picture-iconss>`_ to **True**.
+
+5 - (Optional) If you would like to allow users on your system who are not System Admins to create OAuth 2.0 applications, then set `Restrict managing integrations to Admins <https://docs.mattermost.com/administration/config-settings.html#restrict-managing-integrations-to-admins>`_ to **False**.
+
+Register an OAuth 2.0 Application
+~~~~~~~~~~~~~~~~~~~~~~
+1 - Go to **Main Menu** > **Integrations**
+
+2 - Click **OAuth 2.0 Applications**, then click **Add OAuth 2.0 Application**
+
+3 - Set **Is Trusted**: When set to **Yes**, your application is considered trusted by Mattermost. This means Mattermost doesn't require users to accept authorization when signing to third-party applications. When set to **No**, users will be provided with the following page to accept or deny authorization when authenticating for the first time.
+
+.. image:: ../images/oauth2_authorization_screen.PNG
+
+4 - Specify the **Display Name**: Enter a name for your application made of up to 64 characters. This is the name users will see when granting access to the application, when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications** and when viewing a list of OAuth 2.0 applications in **Main Menu** > **Integrations**.
+
+5 - Add **Description**: This is a short description of your application that users will see when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications**.
+
+6 - Specify the **Homepage**: This is the homepage of the OAuth 2.0 application. The URL must be a valid URL and start with ``http://`` or ``https://`` depending on your server configuration.
+
+7 - (Optional) Add **Icon URL**: The image users will see when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications** and when viewing a list of OAuth 2.0 applications in **Main Menu** > **Integrations**. Must be a valid URL and start with ``http://`` or ``https://``.
+
+8 - Add **Callback URLs**: This are the URL(s) to which Mattermost will redirect users after accepting or denying authorization of your application, and which will handle authorization codes or access tokens. Each URL must be on a separate line and start with ``http://`` or ``https://``.
```

Comment:
Additional note: the Callbacks URL(s) are the only ones that can be used to authorize and app. Mattermost will only redirect to the **one URL** used when authorizing the app